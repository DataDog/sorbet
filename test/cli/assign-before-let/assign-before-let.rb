# typed: true

x = "hello"                   # Error on this line should include a link to GitHub issue
x = T.let("goodbye", String)
