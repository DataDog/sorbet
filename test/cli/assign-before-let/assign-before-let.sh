#!/bin/bash

main/sorbet --silence-dev-message test/cli/assign-before-let/assign-before-let.rb 2>&1
