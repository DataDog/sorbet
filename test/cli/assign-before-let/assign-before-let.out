test/cli/assign-before-let/assign-before-let.rb:3: Incompatible assignment to variable declared via `let`: `String("hello")` is not a subtype of `NilClass` https://srb.help/7001
     3 |x = "hello"                   # Error on this line should include a link to GitHub issue
            ^^^^^^^
    test/cli/assign-before-let/assign-before-let.rb:3: This may be caused by a known Sorbet limitation when `T.let` is in use. See `https://github.com/sorbet/sorbet/issues/3482` for more details.
     3 |x = "hello"                   # Error on this line should include a link to GitHub issue
            ^^^^^^^
Errors: 1
